src/adm/settings.rs

```diff
@@ -17,6 +17,52 @@ use crate::{
 /// is defined for a given partner.
 pub(crate) const DEFAULT: &str = "DEFAULT";
 
+/// The AdvertiserUrlFilter describes the filtering rule for the `advertiser_url`.
+///
+/// Each rule consists of a host and optionally a list of PathFilters.
+///
+/// Examples:
+///
+/// ```json
+///     {
+///         "host": "foo.com",
+///         "paths": [
+///             { "value": "/", "matching": "exact" },
+///             { "value": "/bar/", "matching": "prefix" },
+///             { "value": "/baz/spam/", "matching": "prefix" },
+///         ]
+///     }
+/// ```
+/// For each `advertiser_url` (assume its host is `host` and path is `path`),
+/// the matching rule is defined as follows:
+///
+/// * Check if `host` matches with `filter.host`. If not, this URL is rejected
+///   by this filter.
+/// * If the host matches, and there is no `paths` specified in this filter,
+///   then the URL is accepted by this filter.
+/// * If the `paths` filter list is present, then proceed with path filtering.
+///   There are two "matching" strategies:
+///   * "exact" for exact path matching, which compares the `path` character-by-character
+///     with the `value` filed of this path filter.
+///   * "prefix" for prefix path matching, which checks if the `value` is a
+///     prefix of the `path`. Note that we always make sure `path` and `value`
+///     are compared with the trailing '/' to avoid the accidental
+///     matches. In particular, Contile will panic when it detects that a prefix
+///     filter doesn't have the trailing '/' in the `value`.
```

Comment:
I don't believe that this line is true. Your tests include several "prefix" paths that do not end with slashes and there is no panic.

Reply:
Nice catch, that was inaccurate. It will be only enforced when loading the filters from the configuration file. I will update it.